mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java

```diff
@@ -122,6 +132,66 @@ public void onBackStackChanged() {
+        } else if (actions.trim().equals(getString(R.string.centers))) {
+            replaceFragment(CenterListFragment.newInstance(), false, R.id.container);
+        } else if (actions.trim().equals(getString(R.string.settings))) {
+            startActivity(new Intent(this, SettingsActivity.class));
+        }
+
+    }
+
+    private void initialiseShortcuts() {
+        ShortcutManager shortcutManager;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N_MR1) {
+            shortcutManager = getSystemService(ShortcutManager.class);
+            Intent intent = new Intent(this, SplashScreenActivity.class);
+            List<ShortcutInfo> shortcutList = new ArrayList<>();
+            shortcutList.add(
+                    new ShortcutInfo.Builder(this, getString(R.string.clients))
+                            .setShortLabel(getString(R.string.clients))
+                            .setLongLabel(getString(R.string.clients))
+                            .setIcon(Icon.createWithResource(this, R.drawable.ic_person_black_24dp))
+                            .setIntent(intent.setAction(
+                                    getString(R.string.clients)))
+                            .build()
+            );
+            shortcutList.add(
+                    new ShortcutInfo.Builder(this, getString(R.string.groups))
+                            .setShortLabel(
+                                    getString(R.string.groups))
+                            .setLongLabel(getString(R.string.groups))
+                            .setIcon(Icon.createWithResource(this,
+                                    R.drawable.ic_group_black_24dp))
+                            .setIntent(intent.setAction(
+                                    getString(R.string.groups)))
+                            .build()
+            );
+            shortcutList.add(
+                    new ShortcutInfo.Builder(this, getString(R.string.centers))
+                            .setShortLabel(getString(R.string.centers))
+                            .setLongLabel(getString(R.string.centers))
+                            .setIcon(Icon.createWithResource(this, R.drawable.ic_centers_24dp))
+                            .setIntent(intent.setAction(getString(R.string.centers)))
+                            .build()
+            );
+            shortcutList.add(
+                    new ShortcutInfo.Builder(this, getString(R.string.settings))
+                            .setShortLabel(getString(R.string.settings))
+                            .setLongLabel(getString(R.string.settings))
+                            .setIcon(Icon.createWithResource(this,
+                                    R.drawable.ic_settings))
+                            .setIntent(intent.setAction(getString(R.string.settings)))
+                            .build()
+            );
+            shortcutManager.setDynamicShortcuts(shortcutList);
```

Comment:
Don't you want to do this logic in cycle? Code is repeat five times, instead you could move this strings and drawable consts into two arrays and iterate over it by creating shortcut and adding it into the final collection.